extensions/wikia/Recirculation/services/CuratedContentService.class.php

```diff
@@ -12,32 +12,28 @@ class CuratedContentService {
 	 * Get curated posts. Uses cache if available
 	 * @return an array of posts
 	 */
-	public function getPosts() {
+	public function getPosts( $type, $limit ) {
+		$memcKey = wfSharedMemcKey( __METHOD__, self::MCACHE_VER, $type, $limit );
 
-		$memcKey = wfSharedMemcKey( __METHOD__, self::MCACHE_VER );
-
-		$data = WikiaDataAccess::cache(
+		return WikiaDataAccess::cache(
 			$memcKey,
 			self::MCACHE_TIME,
 			function() {
 				return $this->apiRequest();
 			}
 		);
-
-		return $data;
 	}
 
 	/**
 	 * Make an API request to parsely to gather posts
-	 * @param string $type
 	 * @return an array of posts
 	 */
-	private function apiRequest( $type, $meta ) {
+	private function apiRequest() {
 		$response = ExternalHttp::get( self::API_BASE );
 		$data = json_decode( $response, true );
 
 		if ( isset( $data['posts'] ) && is_array( $data['posts'] ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "posts" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)